linera-service/tests/end_to_end_tests.rs

```diff
@@ -1065,6 +1065,8 @@ async fn test_wasm_end_to_end_non_fungible(config: impl LineraNetConfig) -> Resu
     )
     .await;
 
+    // The process inbox is needed on chain1 in order to get the application.
+    node_service1.process_inbox(&chain1).await?;
```

Comment:
If the `transfer` call above succeeded, it must already have the application. This `process_inbox` really shouldn't be necessary, as discussed.

Reply:
(If removing this really breaks CI, we should keep it, of course, and create an issue to track the bug.)

Reply:
Ok, will see. I need to be convinced that the removal are triggering that problem and I am not convinced so far.